src/stores/fetchSymbolList.ts

```diff
@@ -0,0 +1,42 @@
+import { openDB } from 'idb';
+
+export type SymbolList = {
+  symbol: string;
+  exchange: string;
+  exchangeShortName: string;
+  price: string;
+  name: string;
+};
+
+export const fetchSymbolList = async (): Promise<SymbolList[]> => {
+  const db = await openDB('symbolDB', 1, {
+    upgrade(db) {
+      if (!db.objectStoreNames.contains('symbols')) {
+        db.createObjectStore('symbols', { keyPath: 'symbol' });
+      }
+    },
+  });
```

Comment:
Okay, so a few things. 

1. `fetchSymbolList` should be solely responsible for fetching the data, no DB operations should be defined or called here 
2. For DB operations you should create a separate functions, ideally each for init, and all the CRUD (based on what you need)
3. Getting the symbols flow should be the following in separate hook. You don't need the `SymbolListContext` at all
  a. Check if DB contains the data
  b. If yes, get the data from DB
  c. if no, get the data from useQuery, then add it to db